### ISSUE TITLE
Downloadable product links removal on update via API

### ISSUE THREAD

**Issue by vovsky** [magento/magento2#15923]
### Preconditions
1. Magento 2.2.3/2.2.4
2. Create regular downloadable product, upload downloadable content

### Steps to reproduce
1. Update downloadable product via REST API with call: PUT rest/V1/products/{sku}:
```
{
  "product": {
    "name": "BasicBeginner's Yoga (updated)"
  }
}
```

### Expected result
1. Product attributes specified in request updated, other data remain unchanged

### Actual result
1. After update product has no links to download content (no downloadable information).

<img width="1086" alt="beginner_s_yoga___products___inventory___catalog___magento_admin_after" src="https://user-images.githubusercontent.com/1716535/41085748-8cee6b16-6a40-11e8-8c79-d72c1adbb2e6.png">

**Comment by ghost** [comment]
hi @vovsky  thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by p-bystritsky** [comment]
This issue is fixed in 2.2-dev branch.

**Comment by Ctucker9233** [comment]
@p-bystritsky @magento-engcom-team What commits fix this?

**Comment by sdzhepa** [comment]
Hello @vovsky and @Ctucker9233 

I found the issue was fixed 

Should be available for 2.2.6, see commits for more details:
- 2434fc2d7094692d4541ae4b3a084b46f30f433e

Should be available for 2.3.0 see commits for more details:
- f8a27742d912ec07b5e2d45ba5798c2374e59c69
- fce92ae9d06c2ff73a6e9dc4a37c704930522018


This ticket can be closed.
Please, feel free to reopen or create new one if issue still exists or was not fully fixed

Thank you for feedback and collaboration
